Molio/Source/Domain/UseCase/CurrentUserIdUseCase/DefaultCurrentUserIdUseCase.swift

```diff
@@ -9,12 +9,19 @@ struct DefaultCurrentUserIdUseCase: CurrentUserIdUseCase {
         self.authService = authService
         self.usecase = usecase
     }
-    
+    // TODO: 로그인 재인증 로직 (로그인 화면으로 이동, 재인증 후 다시 불러오기 등) 처리
     func execute() throws -> String? {
         if usecase.isLogin() {
-            return try authService.getCurrentID()
-        } else {
-            return nil
+            do {
+                return try authService.getCurrentID()
+            } catch FirebaseAuthError.userNotFound {
+                print("User not logged in. Redirecting to login.")
+            } catch FirebaseAuthError.requiresReauthentication {
+                print("Token expired. Asking user to reauthenticate.")
+            } catch {
+                print("Unhandled error: \(error)")
+            }
```

Comment:
P1;
`FirebaseAuthError`는 데이터 레이어에 정의된 에러라서, 유즈케이스에서 해당 타입을 알면 안될 것 같습니다.

`FirebaseAuthError`가 `LocalizedError`를 채택하도록 하면, 구체 타입을 몰라도 `error.localizedDescription`로 에러 메세지를 추출할 수 있을 것 같습니다!!
```suggestion
            do {
                return try authService.getCurrentID()
            } catch  {
                print("error.localizedDescription")
            }
```
``` Swift
enum FirebaseAuthError: LocalizedError {
    case loginFailed
    case logoutFailed
    case requiresReauthentication
    case userNotFound
    case deleteAccountFailed
    case reauthenticateFailed
    
    var errorDescription: String? {
        switch self {
        case .loginFailed:
            return "로그인 실패"
        case .logoutFailed:
            return "로그인 아웃 실패"
        case .requiresReauthentication:
            return "재인증이 필요함"
        case .userNotFound:
            return "사용자를 찾을 수 없음"
        case .deleteAccountFailed:
            return "계정 탈퇴 실패"
        case .reauthenticateFailed:
            return "재인증 실패"
        }
    }
}
```